### PR TITLE
feat: add default networkTopology configuration

### PR DESCRIPTION
**Pull request by fcgxz2003**
Signed-off-by: XZ <834756128@qq.com>

## Description

Add default networkTopology configuration.

Struct networkTopology includes parameters: Enable, SyncInterval, CollectInterval, Probe. 
Enable networkTopology services, including probe, network topology collection and synchronization service.
SyncInterval is the interval of synchronizing network topology between schedulers.
CollectInterval is the interval of collecting network topology.
Probe is the configuration of probe.

Struct Probe includes parameters: QueueLength, SyncInterval, SyncCount.
QueueLength is the length of probe queue in directed graph.
SyncInterval is the interval of synchronizing host's probes.
SyncCount is the number of probing hosts.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: scheduler/config/config.go [hunk 1]
```diff
@@ -65,6 +65,9 @@ type Config struct {
 
 	// Network configuration.
 	Network NetworkConfig `yaml:"network" mapstructure:"network"`
+
+	// Probe configuration.
+	Probe ProbeConfig `yaml:"probe" mapstructure:"probe"`
```
Comment (gaius-qi) [inline scheduler/config/config.go:70]:
```golang
NetworkTopology NetworkTopologyConfig `yaml:"networkTopology" mapstructure:"networkTopology"`
```

File: scheduler/config/config.go [hunk 2]
```diff
@@ -299,6 +302,26 @@ type NetworkConfig struct {
 	EnableIPv6 bool `mapstructure:"enableIPv6" yaml:"enableIPv6"`
 }
 
+type ProbeConfig struct {
```
Comment (gaius-qi) [inline scheduler/config/config.go:305]:
```golang
type NetworkTopologyConfig struct {
   Probe ProbeConfig `yaml:"probe" mapstructure:"probe"`
}

type ProbeConfig struct {
}
```

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: scheduler/config/config.go [hunk 3]
```diff
@@ -299,6 +302,31 @@ type NetworkConfig struct {
...
+	// StoreNetworkTopologyInterval is the interval at which the network topology is stored locally.
+	StoreNetworkTopologyInterval time.Duration `mapstructure:"storeNetworkTopologyInterval" yaml:"storeNetworkTopologyInterval"`
+
+	//Probe is the configuration of probe
+	Probe ProbeConfig `yaml:"probe" mapstructure:"probe"`
+}
+
+type ProbeConfig struct {
+	// ProbeQueueLength is the number of probes that an edge stores.
+	ProbeQueueLength int `mapstructure:"probeQueueLength" yaml:"probeQueueLength"`
```
Comment (gaius-qi) [inline scheduler/config/config.go:321]: `ProbeQueueLength` => `QueueLength`

File: scheduler/config/config.go [hunk 4]
```diff
@@ -299,6 +302,31 @@ type NetworkConfig struct {
 	EnableIPv6 bool `mapstructure:"enableIPv6" yaml:"enableIPv6"`
 }
 
+type NetworkTopologyConfig struct {
+	// AOIPriorityInterval is the priority effect of the information of age for host.
+	AOIPriorityInterval time.Duration `mapstructure:"AOIPriorityInterval" yaml:"AOIPriorityInterval"`
```
Comment (gaius-qi) [inline scheduler/config/config.go:307]: Remove `AOIPriorityInterval`.

File: scheduler/config/config.go [hunk 5]
```diff
@@ -299,6 +302,31 @@ type NetworkConfig struct {
...
+
+	// SyncNetworkTopologyInterval is the interval at which network topologies are synchronized between schedulers.
+	SyncNetworkTopologyInterval time.Duration `mapstructure:"syncNetworkTopologyInterval" yaml:"syncNetworkTopologyInterval"`
+
+	// StoreNetworkTopologyInterval is the interval at which the network topology is stored locally.
+	StoreNetworkTopologyInterval time.Duration `mapstructure:"storeNetworkTopologyInterval" yaml:"storeNetworkTopologyInterval"`
```
Comment (gaius-qi) [inline scheduler/config/config.go:313]: `StoreNetworkTopologyInterval` => `CollectInterval`

File: scheduler/config/config.go [hunk 6]
```diff
@@ -299,6 +302,31 @@ type NetworkConfig struct {
...
+	//Probe is the configuration of probe
+	Probe ProbeConfig `yaml:"probe" mapstructure:"probe"`
+}
+
+type ProbeConfig struct {
+	// ProbeQueueLength is the number of probes that an edge stores.
+	ProbeQueueLength int `mapstructure:"probeQueueLength" yaml:"probeQueueLength"`
+
+	// GetProbesInterval is the interval at which the host get the probe list.
+	GetProbesInterval time.Duration `mapstructure:"getProbesInterval" yaml:"getProbesInterval"`
```
Comment (gaius-qi) [inline scheduler/config/config.go:324]: `GetProbesInterval` => `SyncInterval`

File: scheduler/config/config.go [hunk 7]
```diff
@@ -299,6 +302,31 @@ type NetworkConfig struct {
...
+
+	// GetProbeCount is the number of targets that the scheduler sends to the host for probing.
+	GetProbeCount int `mapstructure:"getProbeCount" yaml:"getProbeCount"`
```
Comment (gaius-qi) [inline scheduler/config/config.go:327]: `GetProbeCount` => `SyncCount`

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: scheduler/config/config.go [hunk 8]
```diff
@@ -303,28 +303,25 @@ type NetworkConfig struct {
...
+	CollectInterval time.Duration `mapstructure:"collectInterval" yaml:"collectInterval"`
 
 	//Probe is the configuration of probe
 	Probe ProbeConfig `yaml:"probe" mapstructure:"probe"`
 }
 
 type ProbeConfig struct {
-	// ProbeQueueLength is the number of probes that an edge stores.
-	ProbeQueueLength int `mapstructure:"probeQueueLength" yaml:"probeQueueLength"`
+	// QueueLength is the maximum number of probes that an edge stores.
```
Comment (gaius-qi) [inline scheduler/config/config.go:317]: QueueLength is the length of probe queue in directed graph.

File: scheduler/config/config.go [hunk 9]
```diff
@@ -303,28 +303,25 @@ type NetworkConfig struct {
...
-	// StoreNetworkTopologyInterval is the interval at which the network topology is stored locally.
-	StoreNetworkTopologyInterval time.Duration `mapstructure:"storeNetworkTopologyInterval" yaml:"storeNetworkTopologyInterval"`
+	// CollectInterval is the interval at which the network topology is collected locally.
+	CollectInterval time.Duration `mapstructure:"collectInterval" yaml:"collectInterval"`
 
 	//Probe is the configuration of probe
 	Probe ProbeConfig `yaml:"probe" mapstructure:"probe"`
 }
 
 type ProbeConfig struct {
```
Comment (gaius-qi) [inline scheduler/config/config.go:316]: Add `Enable` params to `ProbeConfig`.

File: scheduler/config/constants.go [hunk 10]
```diff
@@ -146,21 +146,18 @@ const (
...
-
-	// DefaultGetProbesInterval is the default interval at which the host get the probe list.
-	DefaultGetProbesInterval = 30 * time.Second
+	// DefaultSyncNetworkTopologyInterval is the default interval at which network topologies are synchronized between schedulers.
+	DefaultSyncNetworkTopologyInterval = 30 * time.Second
 
-	// DefaultAOIPriorityInterval is the default priority effect of the information of age for host.
-	DefaultAOIPriorityInterval = 10 * time.Second
+	// DefaultCollectInterval is the default interval at which the network topology is collected locally.
+	DefaultCollectInterval = 60 * time.Second
```
Comment (gaius-qi) [inline scheduler/config/constants.go:153]:
```go
// DefaultNetworkTopologyCollectInterval is the default interval of collecting network topology.
DefaultNetworkTopologyCollectInterval = 60 * time.Second
```

File: scheduler/config/constants.go [hunk 11]
```diff
@@ -146,21 +146,18 @@ const (
 )
 
 const (
-	// DefaultProbeQueueLength is the default number of probes that an edge stores.
-	DefaultProbeQueueLength = 5
-
-	// DefaultGetProbesInterval is the default interval at which the host get the probe list.
-	DefaultGetProbesInterval = 30 * time.Second
+	// DefaultSyncNetworkTopologyInterval is the default interval at which network topologies are synchronized between schedulers.
+	DefaultSyncNetworkTopologyInterval = 30 * time.Second
```
Comment (gaius-qi) [inline scheduler/config/constants.go:150]:
```go
// DefaultNetworkTopologySyncInterval is the default interval of synchronizing network topology between schedulers.
DefaultNetworkTopologySyncInterval = 30 * time.Second
```

File: scheduler/config/constants.go [hunk 12]
```diff
@@ -146,21 +146,18 @@ const (
...
 
-	// DefaultAOIPriorityInterval is the default priority effect of the information of age for host.
-	DefaultAOIPriorityInterval = 10 * time.Second
+	// DefaultCollectInterval is the default interval at which the network topology is collected locally.
+	DefaultCollectInterval = 60 * time.Second
 
-	// DefaultGetProbeCount is the default number of targets that the scheduler sends to the host for probing.
-	DefaultGetProbeCount = 50
+	// DefaultQueueLength is the default maximum number of probes that an edge stores.
```
Comment (gaius-qi) [inline scheduler/config/constants.go:155]:
```go
// DefaultProbeQueueLength is the default length of probe queue in directed graph.
DefaultProbeQueueLength = 5
```

File: scheduler/config/constants.go [hunk 13]
```diff
@@ -146,21 +146,18 @@ const (
...
+	DefaultQueueLength = 5
 
-	// DefaultSyncNetworkTopologyInterval is the default interval at which network topologies are synchronized between schedulers.
-	DefaultSyncNetworkTopologyInterval = 30 * time.Second
+	// DefaultSyncProbesInterval is the default interval at which network topology synchronizes the probes of host.
+	DefaultSyncProbesInterval = 30 * time.Second
```
Comment (gaius-qi) [inline scheduler/config/constants.go:159]:
```go
// DefaultProbeSyncInterval is the default interval of synchronizing host's probes.
DefaultProbeSyncInterval = 30 * time.Second
```

File: scheduler/config/constants.go [hunk 14]
```diff
@@ -146,21 +146,18 @@ const (
...
 
-	// DefaultStoreNetworkTopologyInterval is the default interval at which the network topology is stored locally.
-	DefaultStoreNetworkTopologyInterval = 60 * time.Second
+	// DefaultSyncCount is the default number of targets that the scheduler sends to the host for probing.
+	DefaultSyncCount = 50
```
Comment (gaius-qi) [inline scheduler/config/constants.go:162]:
```go
// DefaultProbeSyncCount is the default number of probing hosts.
DefaultProbeSyncCount = 50
```

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: scheduler/config/config.go [hunk 15]
```diff
@@ -299,6 +302,31 @@ type NetworkConfig struct {
 	EnableIPv6 bool `mapstructure:"enableIPv6" yaml:"enableIPv6"`
 }
 
+type NetworkTopologyConfig struct {
+	// Enable networkTopology services, including probe, network topology collection and synchronization service.
```
Comment (gaius-qi) [inline scheduler/config/config.go:306]: `Enable network topology services, including probe, network topology collection and synchronization service.`

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by yxxhero — APPROVED** [review]

**Pull request merged**